### ISSUE TITLE
XForms which use "media::image" have a spinner below the image

### ISSUE THREAD

**Issue by kennsippell** [medic/cht-core#6321]
**Describe the bug**
Elements in XForms which define a `media::image::x` have a random "spinner" which appears below them.

**To Reproduce**

1. Create an XForm with several elements of type "note" which also define `media::image:en`. For example, mRDT workflow for Muso
1. Upload the form and view it

**Behavior**

Observed - There is a random spinner under each note section
Expected - Just the note and the image without the added spinner

**Screenshots**
![Screenshot from 2020-03-22 12-59-10](https://user-images.githubusercontent.com/9014751/77259287-c0b34380-6c3d-11ea-8876-83522812ae40.png)

**Environment**
- Instance: demo-cht.dev
- Browser: Chrome
- Client platform: Bionic
- App: webapp
- Version: 3.7.1 and 3.8.0

**Workaround**
In one of the notes, I'm adding `<style>.pages .loader { display: none; }</style>` which will remove all spinners from the form.

**Comment by garethbowen** [comment]
I cannot reproduce this using a demo form and I can't find the form you used. Can you please link which exact form you are using? I can't find that image anywhere in the medic-projects/muso repo.

I did notice a bug where if the image resource request errors (eg: 404 when the resource isn't attached to the form) then the spinner is left behind instead of showing the error. Is it possible this is the bug here and it's actually a second image that's failing?

**Comment by kennsippell** [comment]
This [Covid-19](https://github.com/medic/config-cht-demo/tree/master/covid19-rdt) form reproduces it. You'll just have to remove the workaround tag that adds the CSS stub.

**Comment by garethbowen** [comment]
Thanks! I've replicated it. I think this is potentially worse than you think... locally I was able to make it fail loading any images at all. I think there is a race condition between rendering the HTML and fetching the images, which if hit, means the spinner is left there permanently _and_ the image is never shown. This differs from your description of the issue which implies that both are shown.

Can you double check your forms to see if there are actually two images there, one which loads and one for which the spinner is never replaced?

**Comment by garethbowen** [comment]
PR created. I'm considering backporting this to 3.8.1 if agree it's more than just a cosmetic patch.

**Comment by kennsippell** [comment]
The DOM for the loader is:

```
<div class="loader"><img lang="fr" data-media-src="images/-" data-itext-id="/covid_rdt/finger_group/generated_note_name_30:label" alt="image" style="visibility: hidden;"></div>
```

So yes, there are two images - one valid (loaded) and one invalid `images/-`. 

But what's up with the `lang="fr"` - is this my french media file for some reason? I see it inside the xml in that repo ...

```
          <text id="/covid_rdt/disinfect_group/generated_note_name_33:label">
            <value form="image">jr://images/-</value>
            <value>-</value>
          </text>
```

But there is no such column for it in the xlsx. I'm new to forms, so not sure ho wmuch of this is normal business.

**Comment by garethbowen** [comment]
Interesting. In my testing I removed all the French translations from the xml to try and get a simple use case. My PR also adds handling so the spinner is hidden when the resource 404s, which will probably solve the visible side effects of this. I don't know why the French translations are being generated but that sounds like a separate (and lower priority) issue. The workaround is to manually delete them from the xml.

**Comment by dianabarsan** [comment]
Ready for AT on `6321-render-images-after-html`. 
I managed to replicate the issue using the form from this [comment](https://github.com/medic/cht-core/issues/6321#issuecomment-602963072) and throttling my browser's CPU to 6x. 

![image](https://user-images.githubusercontent.com/35681649/77458680-446c5c00-6e07-11ea-9896-403cb637d588.png)

**Comment by garethbowen** [comment]
Bumping this to 3.8.1 to get it out sooner.

**Comment by ngaruko** [comment]
LGTM . Is the `French translations`  issue worth raising @garethbowen ?

**Comment by garethbowen** [comment]
Probably.

@kennsippell Can you try and work out where the French translation XML is coming from, and if it's a bug, raise an issue?

**Comment by garethbowen** [comment]
Merged and backported to `3.8.x`
